scipy/interpolate/tests/test_rgi.py

```diff
@@ -335,6 +335,41 @@ def test_fillvalue_type(self):
         RegularGridInterpolator(points, values)
         RegularGridInterpolator(points, values, fill_value=0.)
 
+    @pytest.mark.parametrize("dtype", [np.float32, np.float64])
+    @pytest.mark.parametrize("ndim", [1, 2, 3])
+    @pytest.mark.parametrize("method", ["linear", "nearest"])
+    def test_length_one_axes(self, dtype, ndim, method):
+        # gh-23171: length-1 axes in all dimensions are legal
+        # for all methods parametrized above
+        xi = np.array([0], dtype=dtype)
+        points = (xi, ) * ndim
+        val = dtype(1/7)
+        fill = dtype(1/42)
+        nshape = (1, ) * ndim
+        values = np.array(val, dtype=dtype)
+        values.shape = nshape
+        interp = RegularGridInterpolator(
+            points, values, method=method, bounds_error=False, fill_value=fill
+        )
+
+        # grid point
+        x0 = np.concatenate(points)
+        # all NaN point
+        x_nan = np.array([np.nan, ]*ndim, dtype=dtype)
+        # single NaN point
+        x_nan_0 = np.copy(x0)
+        x_nan_0[0] = np.nan
+        # out-of-bounds point
+        x_oob = np.copy(x0)
+        x_oob[0] += 1
+
+        sample = np.array([x0, x_nan, x_nan_0, x_oob], dtype=dtype)
+        wanted = np.array([val, np.nan, np.nan, fill], dtype=dtype)
+        result = interp(sample)
+        assert_array_equal(result, wanted)
```

Comment:
It may not matter too much, but this test is a fair bit more complex looking than the one in your original ticket--is that just to make it more realistic than the original toy case?

Reply:
These test are meant to expliciltly show the desired behaviour... I added them to be sure that my patch (expressly deleting `out[point] = NAN` clause) would not introduce other regressions.

Reply:
Thank you for making sure of not introducing regressions, this is great! The ask to only split a single "assert four things in one line" into several assertions.

TBH I find it a bit surprising that these `out[point]=NAN` branches can be just removed. Will need to audit test coverage I suppose.

As an aside, this whole 2D fast path should be rewritten at some point, cf gh-22291. This is of course absolutely separate, am only mentioning it in case you'd be interested in working on it after this one :-)

Reply:
> TBH I find it a bit surprising that these `out[point]=NAN` branches can be just removed. Will need to audit test coverage I suppose.

Nan and out-of-bounds are handled by the caller here:
https://github.com/scipy/scipy/blob/4c1a4da8110cc709dc2ff5050d882318fc649e20/scipy/interpolate/_rgi.py#L443-L448
so no need to add a sentinel in `out[point]`.